src/app/packages/open-layer-visualizers/overlays/polyline-visualizer.ts

```diff
@@ -5,87 +5,178 @@ import Style from 'ol/style/style';
 import Vector from 'ol/layer/vector';
 import SourceVector from 'ol/source/vector'
 import Stroke from 'ol/style/stroke';
+import Fill from 'ol/style/fill';
 import MultiPolygon from 'ol/geom/multipolygon';
 import Feature from 'ol/feature';
 import { IVisualizerEntity } from '../../imagery/model/imap-visualizer';
-import {cloneDeep as _cloneDeep} from 'lodash';
+import {cloneDeep as _cloneDeep, get as _get, isNil as _isNil} from 'lodash';
+import { IMap } from '../../imagery/model/imap';
 export const FootprintPolylineVisualizerType = 'FootprintPolylineVisualizer';
 
-export const polylineStyles = {
-	staticPolyline: new Style({
-		stroke: new Stroke({
-			color: '#bd0fe2',
-			width: 5
-		}),
-	})
-};
 
 export class FootprintPolylineVisualizer extends EntitiesVisualizer {
 	hoverLayerSource;
 	hoverLayer: Vector;
 
+	strokeColors = {
+		active: `#27b2cf`,
+		displayed: `#9524ad`,
+		hover: `rgb(211, 147, 225)`,
+		undefined: `rgb(211, 147, 225)`
+	};
+
+	hoverStrokeColors = {
+		active: `#27b2cf`,
+		displayed: `#bd0fe2`,
+		hover: `#bd0fe2`,
+		undefined: `#bd0fe2`
+	};
+
+	interactionsStyle =  new Style({
+		stroke: new Stroke({
+			color: 'transparent',
+			width: 5
+		})
+	});
+
 	constructor(args: any) {
 		super(FootprintPolylineVisualizerType, args);
 		this.fillColor = 'transparent';
 		this.strokeColor = 'rgb(211, 147, 225)';
 		this.containerLayerOpacity = 0.5;
 	}
 
+	onInit(mapId: string, map: IMap) {
+		super.onInit(mapId, map);
+		this.initHoverPolygonLayer();
+		this.subscribers.push(this.syncHoverFeature.subscribe(this.onSyncHoverFeature.bind(this)));
+	}
+
 	createLayer() {
 		super.createLayer();
-		this.initHoverPolygonLayer();
 		this.addSelectInteraction();
+		this.addDoubleClickInteraction();
+
 	}
 
 	initHoverPolygonLayer() {
 		this.hoverLayerSource = new SourceVector();
-		this.hoverLayer = new Vector({source: this.hoverLayerSource});
+		this.hoverLayer = new Vector({
+			source: this.hoverLayerSource,
+			style: (feature: Feature) => {
+				const markClass = this.getMarkClass(feature.getId());
+				const color = this.hoverStrokeColors[markClass];
+				return new Style({
+					stroke: new Stroke({
+						width: 5,
+						color
+					}),
+					fill: new Fill({color: 'rgba(255,255,255,0.4)'})
+				});
+			}
+		});
+		this.hoverLayer.setZIndex(1000);
 		this._imap.mapObject.addLayer(this.hoverLayer);
```

Comment:
don't add layer directly to the map object => add via imap
delete old interactions "onResetView"